### PR TITLE
PICARD-1721: Add $join() script function and associated tests

### PR DESCRIPTION
**Pull request by rdswift**
# Summary

Add a scripting function to join the elements of a multi-value variable into a string using a supplied join phrase. An example might be `$set(artist_string,$join(%artists%,\, ))`

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**: Added a `join_multi()` scripting function to join the elements of a multi-value variable into a string using a supplied join phrase.  Also added associated tests.

# Problem
 Provide a way to quickly and easily join the elements of a multi-value variable into a string, using a custom join phrase between the elements.

* JIRA ticket : PICARD-1721

# Solution

Added a `join()` scripting function to join the elements of a multi-value variable into a string using a supplied join phrase.  Also added associated tests.

# Action

### CODE REVIEW

File: picard/script.py [hunk 1]
```diff
@@ -1066,6 +1066,28 @@ def func_map_multi(parser,
     return separator.join(multi_value)
 
 
+def func_join_multi(parser,
+                       multi,
```
Comment (zas) [inline picard/script.py]: Indentation isn't correct here and below lines

Comment (rdswift) [reply picard/script.py]: I think I have it fixed now.  Sorry about that.

**Comment by phw** [comment]
It probably makes sense to merge #1453 first with the decorator, then adapt this PR to it.

File: picard/script.py [hunk 2]
```diff
@@ -1146,3 +1146,23 @@ def func_map(parser, multi, loop_code, separator=MULTI_VALUED_JOINER):
     if not isinstance(separator, str):
         separator = separator.eval(parser)
     return separator.join(multi_value)
+
+
+@script_function(eval_args=False)
+def func_joinmulti(parser, multi, join_phrase, separator=MULTI_VALUED_JOINER):
```
Comment (phw) [inline picard/script.py]: I would be in favor of using `$join` as the function name here.

Comment (zas) [reply picard/script.py]: I'm in favor of `$join` too

**Comment by zas** [comment]
@rdswift can you rebase and update PR subject/description ? (getting rid of $join_multi / $joinmulti occurences)

**Comment by rdswift** [comment]
Updated and rebased.  Looks like one test is failing, but not due to the commit.  Can someone rerun the tests, or do I need to do something (close and reopen the PR or push another commit)?

**Comment by zas** [comment]
> . Can someone rerun the tests

I did, one build failed due to a network issue, solved now.
Thanks for the rebase ;)

**Pull request merged**